SmbusPIIX4.p

```diff
@@ -493,17 +494,34 @@ public NTSTATUS:ioctl_piix4_port_sel(in[], in_size, out[], out_size) {
     return status;
 }
 
-/// Get the base address of the SMBus controller.
+/// Identify the SMBus controller.
 ///
-/// @param out [0] = Base address of the SMBus controller
-/// @param out_size Must be 1
+/// @param out [0] = Type of the SMBus controller, [1] = I/O Base address, [2] = PCI vendor/device ID, [3] = PCI subsystem vendor/device ID
+/// @param out_size Must be 4
 /// @return An NTSTATUS
-forward NTSTATUS:ioctl_base_addr(in[], in_size, out[], out_size);
-public NTSTATUS:ioctl_base_addr(in[], in_size, out[], out_size) {
-    if (out_size < 1)
+forward NTSTATUS:ioctl_identity(in[], in_size, out[], out_size);
+public NTSTATUS:ioctl_identity(in[], in_size, out[], out_size) {
+    if (out_size < 4)
         return STATUS_BUFFER_TOO_SMALL;
-    
-    out[0] = piix4_smba;
+
+    new NTSTATUS:status;
+
+    new name{8} = "PIIX4";
+    // This isn't great, but it gets around 'warning 229: mixing packed and unpacked array indexing or array assignment'
+    out[0] = name{0} << 56 | name{1} << 48 | name{2} << 40 | name{3} << 32 |
```

Comment:
Same as the other, but you can just implement the other versions (1 to 8) of that macro as well.

Reply:
Oh I guess you’d might not have dealt with multichar constants in C on x86… Usually you write them backwards if you want a certain pattern in memory.